File: elixir/lib/formhandle/commands/whoami.ex
defmodule FormHandle.Commands.Whoami do
  alias FormHandle.{Config, Output}

  def run(opts) do
    config = Config.read() || halt_no_config()

    if opts[:json] do
      Output.json(config)
    else
      Output.heading("FormHandle Config")
      domains = Map.keys(config)

      domains
      |> Enum.with_index()
      |> Enum.each(fn {domain, i} ->
        ep = config[domain]
        IO.puts("  #{domain}")

        Output.table([
          {"handler_id", ep["handler_id"] || ""},
          {"email", ep["email"] || ""},
          {"url", ep["handler_url"] || ""}
        ])

        if i < length(domains) - 1, do: IO.puts("")
      end)

      IO.puts("")
    end
  end

  defp halt_no_config do
    Output.error(~s(No .formhandle config found. Run "formhandle init" first.))
    System.halt(1)
  end
end

File: ruby/lib/formhandle/commands/whoami.rb
# frozen_string_literal: true

module FormHandle
  module Commands
    module Whoami
      def self.run(ctx)
        config = Config.read
        unless config
          Output.error('No .formhandle config found. Run "formhandle init" first.')
          exit 1
        end

        if ctx[:json]
          Output.json(config)
          return
        end

        Output.heading("FormHandle Config")
        domains = config.keys
        domains.each_with_index do |domain, i|
          ep = config[domain]
          puts "  #{domain}"
          Output.table([
            ["handler_id", ep["handler_id"] || ""],
            ["email", ep["email"] || ""],
            ["url", ep["handler_url"] || ""],
          ])
          puts if i < domains.length - 1
        end
        puts
      end
    end
  end
end
